lib/util/cache/global/index.ts

```diff
@@ -30,6 +32,16 @@ export function set(
   return global.renovateCache.set(namespace, key, value, minutes);
 }
 
-export function init(cacheDir: string): void {
-  return fileCache.init(cacheDir);
+export function init(config: RenovateConfig): void {
+  if (config.redisUrl) {
+    redisCache.init(config.redisUrl);
+  } else {
+    fileCache.init(config.cacheDir);
+  }
+}
+
+export async function end(config: RenovateConfig): Promise<void> {
```

Comment:
Thinking of another name for that function. Maybe something like `dispose`?

Reply:
Sure, or maybe "teardown" or "cleanup"? I'm also thinking we should have a global init and teardown/cleanup handler that submodules can post their callbacks to so that things are less coupled.

Reply:
Sounds good